app/src/main/java/bg/zahov/app/ui/exercise/ExercisesScreen.kt

```diff
@@ -0,0 +1,221 @@
+package bg.zahov.app.ui.exercise
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardColors
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import bg.zahov.app.data.model.FilterWrapper
+import bg.zahov.app.data.model.state.ExerciseData
+import bg.zahov.app.data.model.state.ExerciseFlag
+import bg.zahov.app.ui.exercise.filter.FilterDialog
+import bg.zahov.fitness.app.R
+
+@Composable
+fun ExercisesScreen(
+    viewModel: ExerciseViewModel = viewModel(),
+    navigateInfo: () -> Unit,
+    navigateBack: () -> Unit
+) {
+    val uiState by viewModel.exerciseData.collectAsStateWithLifecycle()
+
+    if (uiState.showDialog) {
+        FilterDialog(onDismiss = {
+            viewModel.updateShowDialog(false)
+        })
+    }
+    if (uiState.navigateBack) {
+        LaunchedEffect(Unit) {
+            navigateBack()
+        }
+    }
+
+    ExercisesContent(
+        filterItems = uiState.filters,
+        exerciseItems = uiState.exercises,
+        showButton = uiState.flag != ExerciseFlag.Default,
+        removeFilter = { viewModel.removeFilter(it) },
+        clickExercise = {
+            if (uiState.flag == ExerciseFlag.Default) {
```

Comment:
Extract logic in view model ui should be dumb as possible, not taking business decisions

Reply:
done